src/Qowaiv/Chemistry/CasRegistryNumber.cs

```diff
@@ -0,0 +1,172 @@
+﻿#pragma warning disable S1210
+// "Equals" and the comparison operators should be overridden when implementing "IComparable"
+// See README.md => Sortable
+
+namespace Qowaiv.Chemistry;
+
+/// <summary>
+/// A CAS Registry Number, is a unique numerical identifier assigned by the
+/// Chemical Abstracts Service (CAS), US to every chemical substance described
+/// in the open scientific literature. It includes all substances described
+/// from 1957 through the present, plus some substances from as far back as the
+/// early 1800's.
+/// </summary>
+[DebuggerDisplay("{DebuggerDisplay}")]
+[Serializable]
+[SingleValueObject(SingleValueStaticOptions.All, typeof(long))]
+[OpenApiDataType(description: "CAS Registry Number", type: "string", format: "cas-nr", example: "7732-18-5", pattern: "[1-9][0-9]+\\-[0-9]{2}\\-[0-9]", nullable: true)]
+[OpenApi.OpenApiDataType(description: "CAS Registry Number", type: "string", format: "cas-nr", example: "7732-18-5", pattern: "[1-9][0-9]+\\-[0-9]{2}\\-[0-9]", nullable: true)]
+[TypeConverter(typeof(Conversion.Chemistry.CasRegistryNumberTypeConverter))]
+public readonly partial struct CasRegistryNumber : ISerializable, IXmlSerializable, IFormattable, IEquatable<CasRegistryNumber>, IComparable, IComparable<CasRegistryNumber>
+{
+    /// <summary>Represents an empty/not set CAS Registry Number.</summary>
+    public static readonly CasRegistryNumber Empty;
+
+    /// <summary>Represents an unknown (but set) CAS Registry Number.</summary>
+    public static readonly CasRegistryNumber Unknown = new(long.MaxValue);
+
+    /// <summary>Gets the number of characters of CAS Registry Number.</summary>
+    public int Length => IsEmptyOrUnknown() ? 0 : (int)Math.Ceiling(Math.Log10(m_Value));
```

Comment:
waarom niet string.length?

Reply:
Multiple reasons. First, the `.ToString()` contains two dashes by default. Secondly, this one should be faster.